### PR TITLE
Improvements to the session management closes #28

### LINKED ISSUE #28: Session Management

### PR DESCRIPTION
**Pull request by PauluzzNL**
It' s now possible to get joinable sessions, create or kill sessions and automatically find a given name and start a session if it does not yet exist.

### CODE REVIEW

File: tygron-environment/src/main/java/nl/tudelft/contextproject/tygron/DummyConnection.java [hunk 1]
```diff
@@ -53,7 +54,7 @@ public String callGetEvent(String eventName) {
   }
 
   @Override
-  public String callPostEvent(String eventName, Map<String, String> parameters) {
+  public String callPostEvent(String eventName, JSONArray parameters) {
```
Comment (devbridie) [inline tygron-environment/src/main/java/nl/tudelft/contextproject/tygron/DummyConnection.java:57]: This should probably be abstracted away from JSONArray, perhaps a List?

**Comment by devbridie** [comment]
Have you tried running it? Creating and joining a session doesn't seem to work.
Exception:

```
Exception in thread "main" org.json.JSONException: JSONObject["sessionClientToken"] not found.
    at org.json.JSONObject.get(JSONObject.java:473)
    at org.json.JSONObject.getString(JSONObject.java:654)
    at nl.tudelft.contextproject.tygron.SessionManager.joinSession(SessionManager.java:64)
    at nl.tudelft.contextproject.tygron.SessionManager.createOrFindSessionAndJoin(SessionManager.java:94)
    at nl.tudelft.contextproject.tygron.Connector.<init>(Connector.java:30)
    at nl.tudelft.contextproject.eis.democode.DemoConnector.main(DemoConnector.java:14)
```

File: tygron-environment/src/main/java/nl/tudelft/contextproject/tygron/SessionManager.java [hunk 2]
```diff
@@ -51,14 +50,16 @@ public SessionManager(Connection localApiConnection){
    * @return whether the join was successful or not.
    */
   public boolean joinSession(Session session, int slotId) {
```
Comment (nradenovic93) [inline tygron-environment/src/main/java/nl/tudelft/contextproject/tygron/SessionManager.java:52]: Delete slotId as an argument and use session.getSessionId() instead.

File: tygron-environment/src/main/java/nl/tudelft/contextproject/tygron/SessionManager.java [hunk 3]
```diff
@@ -51,14 +50,16 @@ public SessionManager(Connection localApiConnection){
...
+    dataArray.put("VIEWER");    // My application type: EDITOR, VIEWER, ADMIN, BEAM 
+    dataArray.put("");          // My client address (optional)
+    dataArray.put("Tygron-API-Agent"); //  My client computer name (optional)
+    dataArray.put("");          // My client token for rejoining (optional)
+    
     JSONObject data = apiConnection.callPostEventObject(
-        "services/event/IOServicesEventType/JOIN_SESSION/", dataMap);
+        "services/event/IOServicesEventType/JOIN_SESSION/", dataArray);
     
     session.setClientToken(data.getString("sessionClientToken"));
```
Comment (nradenovic93) [inline tygron-environment/src/main/java/nl/tudelft/contextproject/tygron/SessionManager.java]: Should be data.getJSONObject("client").getString("clientToken") instead

File: tygron-environment/src/main/java/nl/tudelft/contextproject/tygron/SessionManager.java (hunk 2 above)
Comment (PauluzzNL) [inline tygron-environment/src/main/java/nl/tudelft/contextproject/tygron/SessionManager.java:52]: The problem here is that with joining you do not know the slotid 'yet'. So it's either giving it here, or create a session somewhere, set the id manually and then join after all.

**Comment by nradenovic93** [comment]
:+1:

**Comment by devbridie** [comment]
Looks good! :+1: Issue is fixed.

**Pull request merged**